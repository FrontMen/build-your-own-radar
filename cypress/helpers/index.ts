export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
