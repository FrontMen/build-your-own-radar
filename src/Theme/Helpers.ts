const sizes = {
  mobile: 480,
  phablet: 481,
  tablet: 768,
  desktop: 1024,
};

export const MediaQueries = {
  phablet: `(min-width: ${sizes.phablet}px)`,
  tablet: `(min-width: ${sizes.tablet}px)`,
  desktop: `(min-width: ${sizes.desktop}px)`,
};
