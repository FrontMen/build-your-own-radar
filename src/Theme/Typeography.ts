import { css } from 'styled-components';

const getModularScale = (num: number, ratio: number) => Math.pow(num, ratio);

interface msTypes {
  readonly ms: number;
}

export const Type = {
  body: css`
    font-size: ${({ theme }) => getModularScale(1, theme.ModularScale.ratio)}em;
    font-weight: 400;
    line-height: ${({ theme }) => theme.ModularScale.lineHeight[0]};
    font-family: ${({ theme }) => theme.fonts.primary};
  `,
  header: css<msTypes>`
    font-size: ${({ ms, theme }) =>
      getModularScale(ms, theme.ModularScale.ratio)}em;
    font-weight: 400;
    line-height: ${({ theme }) => theme.ModularScale.lineHeight[1]};
    font-family: ${({ theme }) => theme.fonts.primary};
  `,
};
