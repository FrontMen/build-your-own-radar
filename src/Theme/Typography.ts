import { css } from 'styled-components/macro';

export const Typography = {
  body: css`
    font-family: ${props => props.theme.fonts.primary};
    font-size: ${props => props.theme.fontSize[0]}em;
    font-weight: 400;
    line-height: ${props => props.theme.lineHeight[0]};
  `,
  header: css`
    font-family: ${props => props.theme.fonts.primary};
    font-weight: 600;
    line-height: ${props => props.theme.lineHeight[1]};
  `,
};
