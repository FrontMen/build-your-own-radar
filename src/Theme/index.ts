import WebFont from 'webfontloader';

WebFont.load({
  google: {
    families: ['Montserrat', 'Roboto'],
  },
});

export const baseTheme = {
  ModularScale: {
    base: 1,
    ratio: 1.5,
    // TODO: this could be based on the Modular scale at a later point possibly
    lineHeight: [1.65, 1.15],
  },


  space: [0, 4, 8, 16, 32, 64, 128, 256],
  pageWidth: '60em',
  fonts: {
    primary: 'Montserrat, san-serif',
    body: 'Roboto, san-serif',
  },

  pallet: {
    blue: '#526de4',
    indigo: '#532778',
    purple: '#7843ab',
    pink: '#f8bfd1',
    red: '#ff5100',
    orange: '#fd7e14',
    yellow: '#fcea0b',
    green: '#31bd70',
    teal: '#20c997',
    cyan: '#17a2b8',
    white: '#fff',
    gray: '#6c757d',
    grayDark: '#343a40',
    primary: '#526de4',
    secondary: '#767676',
    success: '#31bd70',
    info: '#17a2b8',
    warning: '#fcea0b',
    danger: '#ff5100',
    light: '#f8f9fa',
    dark: '#37383d',
  },
};

export type LightThemeType = {
  readonly [K in keyof typeof lightTheme]: typeof lightTheme[K];
};

export const lightTheme = {
  ...baseTheme,
  colors: {
    body: baseTheme.pallet.grayDark,
    background: baseTheme.pallet.white,
  },
};
