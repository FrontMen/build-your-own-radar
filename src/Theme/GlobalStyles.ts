import { createGlobalStyle } from 'styled-components/macro';
import { LightThemeType } from '.';
import { MediaQueries } from 'Theme/Helpers';

export const GlobalStyle = createGlobalStyle<{ theme: LightThemeType }>`
    *,
    *::before,
    *::after {
    box-sizing: border-box;
    }

    body {
        font-size: 14px;
        font-family: ${props => props.theme.fonts.body};
        line-height: ${props => props.theme.lineHeight};
        color: ${props => props.theme.colors.body};
        background-color: ${props => props.theme.colors.background};
        
        @media (${MediaQueries.phablet}) {
            font-size: 16px;
        }
    }
`;
