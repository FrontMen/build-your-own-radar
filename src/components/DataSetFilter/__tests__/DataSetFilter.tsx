import React from 'react';
import { DataSetFilter } from '../';
import { withAllProviders } from 'test/helpers';
import { storeCreator, filtersStateBuilder } from 'test/builders';
import { lightTheme } from 'Theme';

describe('DataSetFilter', () => {
  const availableDates = ['date1', 'date2'];
  it('should render correctly when dataset is empty', () => {
    const { container, getByTestId, queryByTestId } = withAllProviders(
      <DataSetFilter />,
    );

    expect(getByTestId('dataSetFilter-anchor')).toBeTruthy();
    expect(getByTestId('dataSetFilter-text')).toHaveTextContent('');
    expect(queryByTestId('dataSetFilter-dropdown')).toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when data is there', () => {
    const state = filtersStateBuilder({
      dataSet: {
        availableDates,
        selected: availableDates[0],
      },
    });
    const store = storeCreator(state);
    const { container, getByTestId } = withAllProviders(<DataSetFilter />, {
      store,
    });

    expect(getByTestId('dataSetFilter-text')).toHaveTextContent(
      availableDates[0],
    );
    expect(container).toMatchSnapshot();
  });

  it('should render dropdown when text is clicked', () => {
    const selectedIndex = 0;
    const state = filtersStateBuilder({
      dataSet: {
        availableDates,
        selected: availableDates[selectedIndex],
      },
    });
    const store = storeCreator(state);
    const { container, getByTestId } = withAllProviders(<DataSetFilter />, {
      store,
    });

    getByTestId('dataSetFilter-anchor').click();

    expect(getByTestId('dataSetFilter-dropdown')).toBeTruthy();
    expect(container).toMatchSnapshot();
    availableDates.forEach((date, index) => {
      expect(
        getByTestId(`dataSetFilter-dropdown-option-${date}`),
      ).toHaveStyleRule(
        'background-color',
        index === selectedIndex
          ? lightTheme.pallet.primary
          : lightTheme.pallet.white,
      );
      expect(
        getByTestId(`dataSetFilter-dropdown-option-${date}`),
      ).toHaveTextContent(date);
      expect(
        getByTestId(`dataSetFilter-dropdown-option-${date}`),
      ).toHaveTextContent(date);
    });
  });
});
