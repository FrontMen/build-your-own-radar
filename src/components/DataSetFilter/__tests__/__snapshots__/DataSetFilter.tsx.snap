// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSetFilter should render correctly when data is there 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  cursor: pointer;
  padding: 8px;
  margin-bottom: 20px;
}

.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px 0;
  padding-left: 8px;
  border-bottom: 2px solid #767676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  fill: #767676;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 1.5rem;
  -webkit-transform: scale( 1 );
  -ms-transform: scale( 1 );
  transform: scale( 1 );
}

.c2 {
  margin-right: 8px;
}

@media (min-width:1024px) {
  .c0 {
    margin-left: 0.5em;
  }
}

<div>
  <div
    class="c0"
    data-testid="dataSetFilter-container"
  >
    <div
      class="c1"
      data-testid="dataSetFilter-anchor"
    >
      <span
        class="c2"
        data-testid="dataSetFilter-text"
      >
        date1
      </span>
      <svg
        aria-hidden="true"
        class="c3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`DataSetFilter should render correctly when dataset is empty 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  cursor: pointer;
  padding: 8px;
  margin-bottom: 20px;
}

.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px 0;
  padding-left: 8px;
  border-bottom: 2px solid #767676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  fill: #767676;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 1.5rem;
  -webkit-transform: scale( 1 );
  -ms-transform: scale( 1 );
  transform: scale( 1 );
}

.c2 {
  margin-right: 8px;
}

@media (min-width:1024px) {
  .c0 {
    margin-left: 0.5em;
  }
}

<div>
  <div
    class="c0"
    data-testid="dataSetFilter-container"
  >
    <div
      class="c1"
      data-testid="dataSetFilter-anchor"
    >
      <span
        class="c2"
        data-testid="dataSetFilter-text"
      />
      <svg
        aria-hidden="true"
        class="c3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`DataSetFilter should render dropdown when text is clicked 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  cursor: pointer;
  padding: 8px;
  margin-bottom: 20px;
}

.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px 0;
  padding-left: 8px;
  border-bottom: 2px solid #767676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  list-style: none;
  width: 100%;
  position: absolute;
  box-shadow: 0 0 4px 0 rgba(0,0,0,0.075),0 8px 8px 0 rgba(0,0,0,0.025);
  padding: 4px 0 0;
  background-color: #fff;
  z-index: 1;
  margin: 0;
}

.c5 {
  display: block;
  background-color: #526de4;
  padding: 0.5em 0 0.5em 8px;
  color: #fff;
  border-top: 1px solid #f3f3f3;
}

.c5:hover {
  background-color: #526de4;
}

.c6 {
  display: block;
  background-color: #fff;
  padding: 0.5em 0 0.5em 8px;
  color: inherit;
  border-top: 1px solid #f3f3f3;
}

.c6:hover {
  background-color: #526de41a;
}

.c3 {
  fill: #767676;
  width: 1em;
  height: 1em;
  display: inline-block;
  font-size: 1.5rem;
  -webkit-transform: scale( -1 );
  -ms-transform: scale( -1 );
  transform: scale( -1 );
}

.c2 {
  margin-right: 8px;
}

@media (min-width:1024px) {
  .c0 {
    margin-left: 0.5em;
  }
}

<div>
  <div
    class="c0"
    data-testid="dataSetFilter-container"
  >
    <div
      class="c1"
      data-testid="dataSetFilter-anchor"
    >
      <span
        class="c2"
        data-testid="dataSetFilter-text"
      >
        date1
      </span>
      <svg
        aria-hidden="true"
        class="c3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
    <ul
      class="c4"
      data-testid="dataSetFilter-dropdown"
    >
      <li
        class="c5"
        data-testid="dataSetFilter-dropdown-option-date1"
      >
        date1
      </li>
      <li
        class="c6"
        data-testid="dataSetFilter-dropdown-option-date2"
      >
        date2
      </li>
    </ul>
  </div>
</div>
`;
